oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/AnalyzerModuleConfig.java

```diff
@@ -111,15 +112,15 @@
     @Getter
     private String segmentStatusAnalysisStrategy = FROM_SPAN_STATUS.name();
 
-    private List<Integer> virtualPeers;
+    private Set<Integer> virtualPeers;
```

Comment:
This list is very short and nearly never used, as we usually have all agents for all proxies unless it is a private implemented proxy. Why do you think we need to improve this?

Reply:
[[        analysisListeners.forEach(listener -> {](https://github.com/apache/skywalking/blob/f3b567160ce61675cb692c3417101162d67093de/oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/TraceAnalyzer.java#L79)](https://github.com/apache/skywalking/blob/f3b567160ce61675cb692c3417101162d67093de/oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/TraceAnalyzer.java#L78-L84)

I think it is worthwhile to modify this field to 'Set' in the preceding code because it can be used in a 'foreach' loop. And all we want to do is make sure `componentId` is included in this `Set`, so `Set` would be ideal.

Reply:
This list in 99% of cases is empty. That is why I am asking. HashSet is not always better than List, especially when we face a small and mostly empty list.
That is why I am asking, we don't encourage you to change codes by following general idea w/o real use cases.
SkyWalking is a very complex project, some codes are tricky, even hijacking due to some reason of history, some are real bugs not found before. 
I want to fully understand the background of changing this.